src/PhpConfig.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of composer/xdebug-handler.
+ *
+ * (c) Composer <https://github.com/composer>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace Composer\XdebugHandler;
+
+/**
+ * @author John Stevenson <john-stevenson@blueyonder.co.uk>
+ */
+class PhpConfig
+{
+    /**
+     * Use the original PHP configuration
+     *
+     * @return array PHP cli options
+     */
+    public function useOriginal()
+    {
+        $this->getDataAndReset();
+        return array();
+    }
+
+    /**
+     * Use standard restart settings
+     *
+     * @return array PHP cli options
+     */
+    public function useStandard()
+    {
+        if ($data = $this->getDataAndReset()) {
+            return array('-n', '-c', $data['tmpIni']);
+        }
+
+        return array();
+    }
+
+    /**
+     * Use environment variables to persist settings
+     *
+     * @return array PHP cli options
+     */
+    public function usePersistent()
+    {
+        if ($data = $this->getDataAndReset()) {
+            Process::setEnv('PHPRC', $data['tmpIni']);
+
+            if ($data['scannedInis']) {
+                Process::setEnv('PHP_INI_SCAN_DIR', '');
+            }
+        }
+
+        return array();
+    }
+
+    /**
+     * Returns restart data if available and resets the environment
+     *
+     * @return array|null
+     */
+    private function getDataAndReset()
+    {
+        if ($data = XdebugHandler::getRestartSettings()) {
+            Process::setEnv('PHPRC', $data['phprc']);
+
+            if ($data['scannedInis']) {
```

Comment:
`scannedInis` could be false if `PHP_INI_SCAN_DIR` was set to an empty string in the original environment: value for `PHP_INI_SCAN_DIR` shall be restored regardless of a number of inis scanned.

Reply:
What of these are the reason we *should not* restore `PHP_INI_SCAN_DIR` to the previous value whatever it may be?

Reply:
Because it doesn't need restoring.

Reply:
Despite that though, there would be no harm in restoring it (in case an application has changed it, perhaps). And the same would go for ignoring the `scannedInis` value when we set it.